DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+
+1. Click "Resolve Issue"
+2. Click the "Fix Version/s" dropdown and mark the issue as fixed in
+   the currently unreleased version
+3. Paste "Fixed in commit xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+   into the comment field (replacing the xxx's with the correct hash)
+4. Press the "Resolve" button
+
+That's it.  You can expect someone else to test the change and close
+the JIRA issue so you don't need to, and should not, close the issue
+yourself.
+
+### Development Environment
+
+You will find build instructions in [BUILD.md](BUILD.md) and
+[README.md](README.md).  You will find more instructions how to set up
+your development environment in the Confluence pages [Java
+Setup](https://cwiki.apache.org/confluence/display/DAFFODIL/Java+Setup+and+Notes)
+and [IDEA
+Setup](https://cwiki.apache.org/confluence/display/DAFFODIL/IntelliJ+IDEA+Setup).
+
+### Directory Organization
+
+Daffodil is composed of one top-level project and 20
+subdirectory-level modules (some of these modules contain only
+generators, macros, tests and tutorials).  Daffodil's source code
+repository is organized as follows:
+
+    daffodil/
+    ├── .asf.yaml              - Enables/disables parts of GitHub's UI
+    ├── .gitattributes         - Excludes KEY from source releases
+    ├── .gitignore             - Ignores auto-generated files from source control
+    ├── .sbtopts               - Tells sbt to use 4GB of memory
+    ├── BUILD.md               - Lists Daffodil's build requirements
+    ├── KEYS                   - Lists public keys of Daffodil developers
+    ├── LICENSE                - Licenses Daffodil under the Apache License
+    ├── NOTICE                 - Lists notices affecting LICENSE
+    ├── README.md              - Explains how to get started
+    ├── build.sbt              - Tells sbt how to build Daffodil
+    ├── containers/
+    │   └── release-candidate  - Defines Docker image used to build release
+    ├── daffodil-cli/
+    │   ├── README.md          - Explains how to start Daffodil
+    │   ├── bin.LICENSE        - Contains licenses of Daffodil and subcomponents
+    │   ├── bin.NOTICE         - Contains notices of Daffodil and subcomponents
+    │   ├── build.sbt          - Tells sbt how to build CLI and installers
+    │   └── src/               - Contains CLI source code, tests, scripts, etc.
+    ├── daffodil-core/
+    │   └── src/               - Contains Daffodil's core
+    ├── daffodil-io/
+    │   └── src/               - Contains Daffodil's input/output/charsets
+    ├── daffodil-japi/
+    │   ├── build.sbt          - Tells sbt how to generate javadocs
+    │   └── src/               - Contains Daffodil's Java API
+    ├── daffodil-lib/
+    │   └── src/               - Contains Daffodil's utilities
+    ├── daffodil-macro-lib/
+    │   └── src/               - Defines Daffodil's Scala 2.x macros
+    ├── daffodil-propgen/
+    │   └── src/               - Contains generators to generate more source code
```

Comment:
Thoughts on removing all these `src/` lines and moving the description to the parent directory? Might make this more compact?